### PR TITLE
Update layers.py added max_pool1d class

### PR DESCRIPTION
**Pull request by mtnvdsk**
created a max_pool1d class and a forward method to pass the output from the layer.

### CODE REVIEW

**Comment by Sam-Armstrong** [comment]
Close #15014

**Comment by Sam-Armstrong** [comment]
Hey @mtnvdsk, I've linked this to the corresponding issue - but in future you should do this yourself by adding a comment ‘Close #Issue_number’, as explained here: https://unify.ai/docs/ivy/overview/contributing/the_basics.html

**Comment by Sam-Armstrong** [comment]
@mtnvdsk Please could you add tests for this class in ivy_tests/test_ivy/test_stateful/test_layers.py? You might find this helpful for writing the tests: https://unify.ai/docs/ivy/overview/deep_dive/ivy_tests.html.

Thanks!

**Comment by mtnvdsk** [comment]
Thank for your suggestion sir and from next time I will use the close issue comment 
As you asked me to add test case I added the test case to the ivy_test/test_ivy/test_stateful/test_layers.py

**Comment by Sam-Armstrong** [comment]
@mtnvdsk Thanks for making that change :)

When I run the test you created using the command pytest ivy_tests/test_ivy/test_stateful/test_layers.py::test_maxpool1d_layer the test cases all fail. Please could you try to resolve this?

Thanks

**Comment by mtnvdsk** [comment]
@Sam-Armstrong sir, Modified both the layers.py and test_layers.py as it passed all the test cases. Could you please check for further changes?

File: ivy_tests/test_ivy/test_stateful/test_layers.py [hunk 1]
```diff
@@ -1089,7 +1089,45 @@ def test_sequential_layer(
...
+    class_name,
+    method_name,
+    ground_truth_backend,
+    init_flags,
+    method_flags,
+):
+    input_dtype, x, kernel_size, stride, padding = x_k_s_p
+
+    if len(x[0].shape) == 2:
+        x[0] = np.expand_dims(x[0], axis=2)
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_stateful/test_layers.py:1111]:
```suggestion
```

File: ivy_tests/test_ivy/test_stateful/test_layers.py [hunk 2]
```diff
@@ -1089,7 +1089,45 @@ def test_sequential_layer(
...
 
+    helpers.test_method(
+        ground_truth_backend=ground_truth_backend,
+        init_flags=init_flags,
+        method_flags=method_flags,
+        init_all_as_kwargs_np={
+            "kernel_size": kernel_size[-1],
+            "stride": stride[-1],
+            "padding": padding[-1],
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_stateful/test_layers.py:1120]:
```suggestion
            "kernel_size": kernel_size,
            "stride": stride,
            "padding": padding,
            "data_format": "NCW",
```

Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_stateful/test_layers.py:1120]:
```suggestion
            "kernel_size": kernel_size,
            "stride": stride,
            "padding": padding,
            "data_format": "NCW",
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@mtnvdsk Sorry for the slow reply. All your test cases are still failing for me - I've suggested some changes here. With these changes for me only one of the test cases is still failing. Could you try this out and see if your getting the same? And if so, please can you try to fix any remaining test fails?

Thanks :)

**Comment by Sam-Armstrong** [comment]
@mtnvdsk There seems to be something strange going on with this, for me the test was passing for everything except jax. Not sure what you're seeing?

**Pull request closed without merging**